src/services/modifiedLinesCollector/ModifiedLine.groovy

```diff
@@ -0,0 +1,46 @@
+package services.modifiedLinesCollector
+
+class ModifiedLine {
+
+    public enum ModificationType {
+        Added,
+        Changed,
+        Removed
+    }
+
+    private int number;
+    private String content;
+    private ModificationType type;
+
+    ModifiedLine(int number, String content, ModificationType type) {
+        this.number = number;
+        this.content = content;
+        this.type = type;
+    }
+
+    public int getNumber() {
+        return number;
+    }
+
+    public String getContent() {
+        return content;
+    }
+
+    public ModificationType getType() {
+        return type;
+    }
+
+    public String toString() {
+        return number + "-" + content + "-" + type
+    }
+
+    @Override
+    public boolean equals(Object modifiedLine) {
```

Comment:
@rafaelmotaalves couldn't this be imprecise?

Reply:
yes, but in this case, the equals method is used to compare merge lines with other revisions lines, so I can't use the line number, because the lines can be in different places for each revision

Reply:
I understand. But this could lead to errors in method bodies that contain repeated lines, right? 

As in 

x=x+1;
y=m();
x=x+1;

If this is the case, we should carefully assess (how frequent are such errors? can we discard such scenarios from the analysis?) and document that, as it might impact the study results. We should do the same for any imprecision that might impact the study results.

I wonder if we could do better by, for example, recording the line contents and, for lines that appear more than once, the order in which they appear. For the example above, we would record (x=x+1,1) for the first line, (y=m(),0) for the second line, and (x=x+1,2) for the third line. And maybe this extra information could be used to eliminate or reduce imprecision.

Reply:
yes, I don't think this case is very common

Reply:
currently, lines with the same content would not be added to the modified line lists, because they are Set. Maybe because both lines do the same thing, this would not impact the result, because they would have the same flows.

Reply:
recording the order which they appear could work, but would we have a problem if both left and right added the same line multiple times?

Reply:
> currently, lines with the same content would not be added to the modified line lists, because they are Set. 

@rafaelmotaalves This is a problem. I will open a new issue. For now, we should discard scenarios with mutually modified methods that have repeated lines. 

> Maybe because both lines do the same thing, this would not impact the result, because they would have the same flows.

No. That's not the case. The flows are often not the same.